src/data/ability.ts

```
@@ -982,29 +984,67 @@ export class FieldPreventExplosiveMovesAbAttr extends AbAttr {
 }
 
 export class MoveTypeChangeAttr extends PreAttackAbAttr {
-  private newType: Type;
-  private powerMultiplier: number;
-  private condition: PokemonAttackCondition;
-
-  constructor(newType: Type, powerMultiplier: number, condition: PokemonAttackCondition) {
+  constructor(
+    private newType: Type,
+    private powerMultiplier: number,
+    private condition?: (move: Move) => boolean
+  ) {
     super(true);
-    this.newType = newType;
-    this.powerMultiplier = powerMultiplier;
-    this.condition = condition;
   }
 
   applyPreAttack(pokemon: Pokemon, passive: boolean, defender: Pokemon, move: PokemonMove, args: any[]): boolean {
-    if (this.condition(pokemon, defender, move.getMove())) {
+    if (this.condition && this.condition(move.getMove())) {
       const type = (args[0] as Utils.IntegerHolder);
       type.value = this.newType;
-      (args[1] as Utils.NumberHolder).value *= this.powerMultiplier;
+
+      if (args[1] && args[1] instanceof Utils.NumberHolder) {
+        args[1].value *= this.powerMultiplier;
+      }
       return true;
     }
 
     return false;
   }
 }
```

Why is this being changed when it's not part of protean/libero?

---

This is being changed for the same reason as I mentioned in a different comment where I invoke this class in order to figure out what type the pokemon needs to change to but in order to do so without causing a run time error I need to pass a NumberHolder as `args[1]` so this is my way of preventing that error.

Also the change to the condition was warranted (at least in my mind) because the previous condition function had unused variables for all of the abilities I had seen with that Attr

---

I mean, the previous condition had unused arguments because it was following the standard that abilities which modify attacks were consistently following. Yes that meant some abilities had more data than they needed, but I'm not sure I'd call that a bad thing either.

I'm also suspicious this might not be able to conceal other things, because this is really just letting a function fail silently when given invalid arguments. Though this will probably still log an activation of the function. I guess my simulated call stuff should be prioritised a bit higher so things like this can work properly.

---

> This is really just letting a function fail silently when given invalid arguments.

The function doesn't fail if the arguments are invalid it just doesn't mutate whatever is in `args[1]` if there is nothing there. The NumberHolder that I'm omitting is normally used as a damage multiplier and since I don't need that to determine the resulting move type I would simply be passing useless information into this class. 

> I mean, the previous condition had unused arguments because it was following the standard that abilities which modify attacks were consistently following. 

I could change it so that the condition is the same as it was before without causing any breaking changes.